entry_types/scrolled/spec/controllers/pageflow_scrolled/editor/chapters_controller_spec.rb

```diff
@@ -0,0 +1,176 @@
+require 'spec_helper'
+require 'pageflow/editor_controller_test_helper'
+
+module PageflowScrolled
+  RSpec.describe Editor::ChaptersController, type: :controller do
+    include Pageflow::EditorControllerTestHelper
+
+    routes { PageflowScrolled::Engine.routes }
+
+    describe '#create' do
+      it 'requires authentication' do
+        entry = create(:entry)
+
+        post(:create,
+             params: {
+               entry_id: entry,
+               chapter: attributes_for(:scrolled_chapter)
+             }, format: 'json')
+
+        expect(response.status).to eq(401)
+      end
+
+      it 'succeeds for authorized user' do
+        entry = create(:entry)
+
+        authorize_for_editor_controller(entry)
+        post(:create,
+             params: {
+               entry_id: entry,
+               chapter: attributes_for(:scrolled_chapter)
+             }, format: 'json')
+
+        expect(response.status).to eq(201)
+      end
+
+      it 'allows setting the chapters configuration hash' do
+        entry = create(:entry)
+
+        authorize_for_editor_controller(entry)
+        post(:create,
+             params: {
+               entry_id: entry,
+               chapter: {
+                 configuration: {title: 'A chapter title'}
+               }
+             }, format: 'json')
+
+        chapter = Chapter.all_for_revision(entry.draft).first
+        expect(chapter.configuration).to eq('title' => 'A chapter title')
+      end
+    end
+
+    describe '#update' do
+      it 'allows updating the chapters configuration hash' do
+        entry = create(:entry)
+        chapter = create(:scrolled_chapter, revision: entry.draft)
+
+        authorize_for_editor_controller(entry)
+        patch(:update,
+              params: {
+                entry_id: entry,
+                id: chapter,
+                chapter: {
+                  configuration: {title: 'A chapter title'}
+                }
+              }, format: 'json')
+
+        expect(chapter.reload.configuration).to eq('title' => 'A chapter title')
+      end
+
+      it 'does not allow updating a chapter from a different entry' do
+        entry = create(:entry)
+        create(:scrolled_chapter, revision: entry.draft)
+        other_entry = create(:entry)
+        chapter_in_other_entry = create(:scrolled_chapter, revision: other_entry.draft)
+
+        authorize_for_editor_controller(entry)
+        expect {
+          patch(:update,
+                params: {
+                  entry_id: entry,
+                  id: chapter_in_other_entry,
+                  chapter: {
+                    configuration: {title: 'another title'}
+                  }
+                }, format: 'json')
+        }.to raise_error(ActiveRecord::RecordNotFound)
```

Comment:
This means a 500 response which is not what we exepct. `Pageflow::ApplicationController` had a `rescue_from` to turn all `ActiveRecord::RecordNotFound` errors into 404s. This sometimes lead to hard to track down bugs if an unexpected exception was silently converted into a 404. `Pageflow::EditorController` therefore rescues these exceptions explicitly at places where they are expected. I'd suggest to the same in the actions here.